nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/test/java/org/nuxeo/ecm/automation/core/operations/document/CopyDocumentTest.java

```diff
@@ -0,0 +1,128 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nour AL KOTOB
+ */
+package org.nuxeo.ecm.automation.core.operations.document;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.Serializable;
+import java.util.Map;
+
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.automation.AutomationService;
+import org.nuxeo.ecm.automation.OperationContext;
+import org.nuxeo.ecm.automation.OperationException;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.event.CoreEventConstants;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+import javax.inject.Inject;
+
+/**
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features(CoreFeature.class)
+@Deploy("org.nuxeo.ecm.automation.core")
+public class CopyDocumentTest {
+    public static final String ROOT = "/";
+
+    public static final String COPY_DOC_NAME = "copyDoc";
+
+    public static final String TARGET_PROPERTY_KEY = "target";
+
+    public static final String NAME_PROPERTY_KEY = "name";
+
+    public static final String FILE = "File";
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected AutomationService automationService;
+
+    @Inject
+    protected TransactionalFeature txFeature;
+
+    @After
+    public void tearDown() {
+        session.removeChildren(session.getRootDocument().getRef());
+        session.save();
+    }
+
+    @Test
+    public void testCopyChildren() throws OperationException {
+        DocumentModel folder = session.createDocumentModel(ROOT, "SourceFolder", "Folder");
+        folder = session.createDocument(folder);
+
+        DocumentModel childDoc = session.createDocumentModel(folder.getPathAsString(), "childDoc", FILE);
+        session.createDocument(childDoc);
+
+        OperationContext context = new OperationContext(session);
+        context.setInput(folder);
+
+        Map<String, Serializable> params = Map.of(TARGET_PROPERTY_KEY, ROOT, NAME_PROPERTY_KEY, "CopyFolder");
+        DocumentModel result = (DocumentModel) automationService.run(context, CopyDocument.ID, params);
+
+        result = session.getDocument(result.getRef());
```

Comment:
You probably should check that:
1/ `result` is not the `folder` document
2/ `result` has the expected name
3/ `result` child is not the same document as `folder` child